ledger/src/blockstore.rs

```diff
@@ -2922,37 +2924,44 @@ impl Blockstore {
         ending_slot: Slot,
     ) -> DashSet<Pubkey> {
         let result = DashSet::new();
+        let lookup_tables = DashSet::new();
 
         (starting_slot..=ending_slot)
             .into_par_iter()
             .for_each(|slot| {
                 if let Ok(entries) = self.get_slot_entries(slot, 0) {
                     entries.into_par_iter().for_each(|entry| {
                         entry.transactions.into_iter().for_each(|tx| {
+                            for account in tx.message.static_account_keys() {
+                                result.insert(*account);
+                            }
+
                             if let Some(lookups) = tx.message.address_table_lookups() {
                                 lookups.iter().for_each(|lookup| {
-                                    result.insert(lookup.account_key);
+                                    lookup_tables.insert(lookup.account_key);
                                 });
                             }
-                            // howdy, anybody who reached here from the panic messsage!
-                            // the .unwrap() below could indicate there was an odd error or there
-                            // could simply be a tx with a new ALT, which is just created/updated
-                            // in this range. too bad... this edge case isn't currently supported.
-                            // see: https://github.com/solana-labs/solana/issues/30165
-                            // for casual use, please choose different slot range.
-                            let sanitized_tx = bank.fully_verify_transaction(tx).unwrap();
-                            sanitized_tx
-                                .message()
-                                .account_keys()
-                                .iter()
-                                .for_each(|&pubkey| {
-                                    result.insert(pubkey);
-                                });
                         });
                     });
                 }
             });
 
+        // For each unique lookup table add all accounts to the minimized set.
+        lookup_tables.into_par_iter().for_each(|lookup_table_key| {
+            let Some(lookup_table_account) = bank.get_account(&lookup_table_key) else {
+                return;
+            };
+
+            let Ok(lookup_table) = AddressLookupTable::deserialize(lookup_table_account.data())
+            else {
+                return;
+            };
+
+            for address in lookup_table.addresses.iter() {
+                result.insert(*address);
```

Comment:
i wish `ExtendLookupTable` accepted addresses in the form of account references, not instruction data with serialized `Vec<Pubkey>`....

if so, things would be simpler...

Reply:
Oh didn't realize it was through ix data.
I can scan for extend ixs and manually add those as well

Reply:
> i wish ExtendLookupTable accepted addresses in the form of account references, not instruction data with serialized Vec<Pubkey>....

Agree this would make things much simpler (for us). But I can understand why it was not doen that way - since there's no need to load those, possibly large, accounts to add them to an ALT.

Reply:
> how about plumbing a flag --non-strict-alt-scanning and panic! as before by default with mention of the flag?

I think these are common enough that I'd like to consider some alternatives first, because otherwise users will use default and hit this often, or have to run twice.

I'm more in favor of doing our best, and logging a warning if we detect a possible issue...which I think is possible?

What do you think about this:
1. We manually scan for "simple" extension instructions
  - any txs w/ native-only programs/ixs
  - we can directly rip pubkeys from the extension ix data  
3. We try detecting potential CPI:
  - even if they CPI the ALT extension, the tx must have the ALT program as an input
  - if we see some tx which uses non-native programs & has the ALT program this means it's possible it will CPI to extend.
  - if we detect this case, we log some warning (at the end) with a description
  
I think this makes the usage a lot easier and robust than it currently is. And it still allows us to print a warning that it may not work, rather than just silently not working.
I'm not sure how common CPI extensions are, but I think we certainly can do better for "simple" ALT create & extends.

Reply:
> @ryoqun what do you think about this:
> ...

sounds nice. a little concern is that people won't notice those warnings as `ledger-tool create-snapshot` is already quite verbose. maybe we can arrange to display the warning at the very last of command termination for maximum discoverability as you said.